### PR TITLE
[DOC] Add subpages for query editor doc

### PR DESCRIPTION
**Pull request by knylander-grafana**
**What this PR does**:

Add subpages for query editor content to match Grafana: https://grafana.com/docs/grafana/latest/datasources/tempo/query-editor/

Original content from this PR: https://github.com/grafana/grafana/pull/74239

This change unifies the query editor and builder content to the Grafana repository under docs/shared/datasource/tempo.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by knylander-grafana** [comment]
I added changes @jdbaldry made to the docs Make file to my PR. It wasn't building locally. His changes resolved the issues with the shared files.

**Review by BeverlyJaneJ — APPROVED** [review]

File: docs/make-docs [hunk 1]
```diff
@@ -626,7 +636,7 @@ ${PODMAN} run \
   ${DOCS_IMAGE} \
   /entrypoint
 EOF
-    await_build http://localhost:3002 &
+    await_build http://localhost:3003 &
```
Comment (mapno) [inline docs/make-docs:639]: This needs a review from someone with more context, possibly @jdbaldry?

**Review by mapno — COMMENTED** [review]
LGTM. I'd like confirmation of the port change. If that's ok, I'll approve and merge.

File: docs/make-docs (hunk 1 above)
Comment (jdbaldry) [reply docs/make-docs:639]: These are my changes and they are expected, thanks for double checking!

Comment (mapno) [reply docs/make-docs:639]: Awesome, thanks

**Review by mapno — APPROVED** [review]

**Pull request merged**